packages/validator/src/commands/utils.ts

```diff
@@ -49,3 +66,68 @@ export function getPath(template: string, messages: Log[], version = ''): string
 
   return res;
 }
+
+/**
+ * Returns path to conversion template between specified formats
+ * @param from - name of source specification format
+ * @param to - name of target specification format
+ * @param messages - array into which warnings/errors will pushed
+ * @param fromVersion (optional) - version of source specification format, if not specified, latest version of the template is returned
+ * @param toVersion (optional) - version of target specification format, if not specified, latest version of the template is returned
+ */
+export function getConversionPath(from: string, to: string, messages: Log[], fromVersion = '', toVersion = ''): string {
+  if (!conversions[from]) {
+    messages.push({ level: 'error', message: `Unknown conversion from ${from} to ${to}` });
+    return '';
+  }
+
+  if (!fromVersion) {
+    let fromLatest;
+
+    for (const version in conversions[from]) {
+      if (!conversions[from][version][to]) {
+        continue;
+      }
+
+      fromLatest = !fromLatest || (fromLatest < version && version.match(/^[0-9\.]+$/)) ? version : fromLatest;
+    }
+
+    if (!fromLatest) {
+      messages.push({ level: 'error', message: `Unknown conversion from ${from} to ${to}` });
+      return '';
+    }
+
+    fromVersion = fromLatest;
+  }
+
+  if (!conversions[from][fromVersion][to]) {
+    messages.push({ level: 'error', message: `Unknown conversion from ${from} to ${to}` });
+    return '';
+  }
+
+  if (!toVersion) {
+    let toLatest;
+
+    for (const version in conversions[from][fromVersion][to]) {
+      toLatest = !toLatest || (toLatest < version && version.match(/^[0-9\.]+$/)) ? version : toLatest;
+    }
+
+    if (!toLatest) {
+      messages.push({ level: 'error', message: `Unknown conversion from ${from} to ${to}` });
+      return '';
+    }
+
+    toVersion = toLatest;
+  }
```

Comment:
This is something I can't decide if it's a correct/desired behavior or not.

When you don't pass a version to `from` and `to` arguments it's expected to choose the latest one. But, what is the latest one? Imagine that we support these conversions:

`apple@1.0.0` -> `orange@1.0.0`
`apple@1.0.0` -> `blueberry@1.0.0`
`apple@1.0.0` -> `banana@1.0.0`
`apple@2.0.0` -> `orange@2.0.0`
`apple@2.0.0` -> `blueberry@2.0.0`

When moving to version `2.0.0` we've droped the conversion from `apple` to `banana`. So currently the latest version of the "product" is `2.0.0`. But if you call `convert --from apple --to banana` it will currently still run but using version `1.0.0` because that's the latest that can do the conversion. Even though there's already a newer version of `apple` available, it just doesn't support conversion to `banana`.

I'd say this is a bit confusing for the user. I, as a user, would expect the message saying that the latest version of `apple` can't be converted to `banana`. And if I'm ok with the older version saying so explicitly.

WDYT?

Reply:
Now I understood, in f8ab06605f0b989f68ec3d055cc6bb368ba553d0 it will issue a warning if the latest version does not have the conversion.